store/event.go

```diff
@@ -0,0 +1,146 @@
+/*
+Copyright 2016 caicloud authors. All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package store
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/caicloud/cyclone/api"
+
+	mgo "gopkg.in/mgo.v2"
+	"gopkg.in/mgo.v2/bson"
+)
+
+const (
+	// retryThreshold represents the threshold for retry time of event.
+	// If the retry time is not more than this threshold, the retry interval will be retry * retryInterval.
+	// the retry interval will be retryThreshold * retryInterval.
+	retryThreshold = 10
+
+	// retryInterval represents the interval unit for retry.
+	retryInterval = 1 * time.Minute
+)
+
+var (
+	// outTimeout represents the timeout for event to keep "out" status.
+	// After the event is out, then its status will be "handling" after it starts to be handled,
+	// otherwise, it will be re-enqueued and its status will change back to "in".
+	outTimeout, _ = time.ParseDuration("-3m")
```

Comment:
Is there any reason for you to use time.ParseDuration? It seems that you exactly know the timeout duration.

Reply:
Fixed